cli/src/commands/git/push.rs

```diff
@@ -295,8 +300,44 @@ pub fn cmd_git_push(
         return Ok(());
     }
 
-    validate_commits_ready_to_push(ui, &bookmark_updates, &remote, &tx, args)?;
+    let sign_behavior = if tx.settings().get_bool("git.sign-on-push")? {
+        Some(SignBehavior::Own)
+    } else {
+        None
+    };
+    let commits_to_sign =
+        validate_commits_ready_to_push(ui, &bookmark_updates, &remote, &tx, args, sign_behavior)?;
+    let num_updated_signatures;
+    let num_rebased_descendants: usize;
+    match (args.dry_run, sign_behavior) {
+        (false, Some(sign_behavior)) => {
+            num_updated_signatures = commits_to_sign.len();
+            (num_rebased_descendants, bookmark_updates) = sign_commits_before_push(
+                &mut tx,
+                commits_to_sign,
+                sign_behavior,
+                bookmark_updates,
+            )?;
+        }
+        _ => {
+            num_updated_signatures = 0;
+            num_rebased_descendants = 0;
+        }
+    }
+
     if let Some(mut formatter) = ui.status_formatter() {
+        if num_updated_signatures > 0 {
+            writeln!(
+                formatter,
+                "Updated signatures of {num_updated_signatures} commits"
+            )?;
+            if num_rebased_descendants > 0 {
+                writeln!(
+                    formatter,
+                    "Rebased {num_rebased_descendants} descendant commits"
+                )?;
+            }
```

Comment:
optional: Move this messages inside the `match` block above and reduce scope of `num_updated_signatures` and `num_rebased_descendants` since they can only be non-zero there. Then the `match` can also become an `if let`. The `if let Some(mut formatter) = ui.status_formatter()` will be duplicated, though.